src/dq_suite/output_transformations.py

```diff
@@ -131,53 +144,53 @@ def extract_dq_afwijking_data(
     :param catalog_name:
     :param spark_session:
     """
-    # Extracting information from the JSON
-    run_time = dq_result["meta"]["run_id"].run_time  # Access run_time attribute
-    # Extracted data for df
+    tabel_id = f"{dataset_name}_{table_name}"
+    run_time = dq_result["meta"]["run_id"].run_time  # Get the run timestamp
     extracted_data = []
-
-    # To store unique combinations of value and IDs
-    unique_entries = set()
+    if type(unique_identifier) is not list: unique_identifier = [unique_identifier]
+    number_of_unique_ids = len(unique_identifier)
 
     for result in dq_result["results"]:
         expectation_type = result["expectation_config"]["expectation_type"]
         parameter_list = create_parameter_list_from_results(result=result)
-        attribute = result["expectation_config"]["kwargs"].get("column")
+        if "column" in result["expectation_config"]["kwargs"]:
+            attribute = result["expectation_config"]["kwargs"].get("column")
+        else:
+            attribute = result["expectation_config"]["kwargs"].get("column_list")
         afwijkende_attribuut_waarde = result["result"].get(
             "partial_unexpected_list", []
         )
-        for value in afwijkende_attribuut_waarde:
+        unieke_afwijkende_waardes = set()
+        for waarde in afwijkende_attribuut_waarde:
+            if isinstance(waarde, dict):
+                waarde = tuple(waarde.items())
+            unieke_afwijkende_waardes.add(waarde)
```

Comment:
Klopt het hier dat je `waarde` ook aan de set wilt toevoegen als het geen `dict` is?

Reply:
Ja, alleen kan een dict blijkbaar niet worden toegevoegd aan een set. Vandaar de voorwaardelijke transformatie. Heb een comment toegevoegd om het te verduidelijken